File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        size = len(nums)
        ans = size + 1
        left = 0
        right = 0
        window_sum = 0

        while right < size:
            window_sum += nums[right]

            while window_sum >= target:
                ans = min(ans, right - left + 1)
                window_sum -= nums[left]
                left += 1

            right += 1

        return ans if ans != size + 1 else 0
            

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0;
        int right = 0;
        int total = 0;
        int ans = INT32_MAX;
        for(; right < nums.size(); right++){
            total += nums[right];
            while (total >= target){
                int length = right - left + 1;
                ans = ans < length ? ans : length;
                total -= nums[left++];
            }
        }
        return ans == INT32_MAX ? 0 : ans;        
    }
};